### PR TITLE
[ArgumentValueResolverToValueResolverRector] Add rule (#289)

### LINKED ISSUE #289: [SYMFONY62] Replace `supports` with `resolve`
Make sure to write a custom rule for this as return type has changed from `bool` to `array` more information can be found on the [symfony pull request](https://github.com/symfony/symfony/pull/47363)

Some more info https://github.com/rectorphp/rector-symfony/pull/279#discussion_r1029400492 & https://github.com/rectorphp/rector-symfony/pull/287

### PR DESCRIPTION
**Pull request by JohJohan**
Fixes #289

### CODE REVIEW

**Comment by stefantalen** [comment]
Don't forget to update `config/sets/symfony/symfony62.php` by adding this rule and remove the `RenameClassRector` lines.

File: src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            }
+        }
+
+        // If it doesn't implement ArgumentValueResolverInterface, skip
+        if (! $isArgumentValueResolver) {
+            return null;
+        }
+
+        $isIdentical = true;
+        foreach ($node->getMethods() as $key => $classMethod) {
```
Comment (johanadivare) [inline src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php:54]: This makes it work but right now its a bit messy and PHPstan also doesn't like i should have a look how to improve this part.

Comment (johanadivare) [reply src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php:54]: @stefantalen maybe you have some suggestions for me? :)

Comment (stefantalen) [reply src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php:54]: Not really, maybe moving some logic to _private_ functions would help.

Comment (johanadivare) [reply src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php:44]: Yeah thanks for the tip i did that :)

**Comment by JohJohan** [comment]
@samsonasik its ready to be reviewed now.

File: src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+        return [$isIdentical, $supportFirstArg, $supportSecondArg];
+    }
+
+    private function processResolveMethod(mixed $classMethod, mixed $isIdentical, mixed $supportFirstArg, mixed $supportSecondArg): void
+    {
+        $ifCondition = $isIdentical ? new NotIdentical($supportFirstArg, $supportSecondArg)
+            : new Identical($supportFirstArg, $supportSecondArg);
+
+        assert(is_array($classMethod->getStmts()));
```
Comment (samsonasik) [inline src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php:144]: here as well, if already checked, passs `$clasMethod->stmts` to `processResolveMethod()` method as parameter can reduce assert check.

File: src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+     * @param ClassMethod $classMethod
+     *
+     * @return array{bool, Expr|null, Expr|null}
+     */
+    private function extractSupportsArguments(Class_ $node, int $key, ClassMethod $classMethod): array
+    {
+        $isIdentical = true;
+        $supportFirstArg = $supportSecondArg = null;
+
+        assert(is_array($classMethod->getStmts()));
```
[the remark this answers is not in the document]
Comment (johanadivare) [reply src/Rector/ClassMethod/ArgumentValueResolverToValueResolverRector.php:118]: Good point, checked on null value

**Comment by johanadivare** [comment]
@samsonasik can you have another look when you have time?

**Review by samsonasik — APPROVED** [review]
Looks good to me

**Review by samsonasik — CHANGES_REQUESTED** [review]
@JohJohan oh, the Rector rule seems moved to `rules` directory, could you move it to sync with latest main branch? thank you.

https://github.com/rectorphp/rector-symfony/tree/main/rules

**Comment by samsonasik** [comment]
btw, @JohJohan @johanadivare are you same person 😄 ?

**Comment by johanadivare** [comment]
> btw, @JohJohan @johanadivare are you same person 😄 ?

Haha yes, @johanadivare is my work Github ;p

**Comment by johanadivare** [comment]
@samsonasik i moved the files to the correct folders

**Comment by samsonasik** [comment]
Thank you @johanadivare , @TomasVotruba lets merge it ;)

**Comment by TomasVotruba** [comment]
Thanks a lot @johanadivare :+1:

**Pull request merged**